src/modalities/models/coca/coca_model.py

```diff
@@ -123,14 +121,6 @@ def __init__(
             attention_config=text_decoder_config.attention_config,
         )
 
-        # init all weights
-        assert weight_init.type in ["plain", "scaled"], f"ERROR! weight_init.type = {weight_init.type} not implemented."
-        self.initialize_weights(

```

Comment:
TODO: check how we want to integrate coca into this

Reply:
[Here](https://github.com/Modalities/modalities/blob/bb4cd363a619792c8d00ccb78416893d0693b354/src/modalities/nn/weight_init/high_level_weight_init_factory.py#L30) we state that only the `plain` initialization can be used for CoCa. I think that is a good starting point. There are just two things missing in my opinion: 
1. We need to exclude the use of `std = 'auto'` for CoCa (the hidden dimension `d` used in `'auto'` mode might not be well-defined as the text and multimodal encoder can in general have different `d`)
2. The unit tests need to be updated.

Reply:
Regarding 1., I think we don't have to "exclude" `std='auto'` because, as you said, it is not applicable and one wouldn't specify `hidden_dim` in the config in the first place. 

Yes, unit tests need to be updated.